Mage/src/main/java/mage/abilities/keyword/PlotAbility.java

```diff
@@ -99,11 +114,12 @@ public static boolean doExileAndPlotCard(Card card, Game game, Ability source) {
         UUID exileId = PlotAbility.getPlotExileId(owner.getId(), game);
         String exileZoneName = "Plots of " + owner.getName();
         Card mainCard = card.getMainCard();
+        Zone zone = game.getState().getZone(mainCard.getId());
         if (mainCard.moveToExile(exileId, exileZoneName, source, game)) {
             // Remember on which turn the card was last plotted.
             game.getState().setValue(PlotAbility.getPlotTurnKeyForCard(mainCard.getId()), game.getTurnNum());
             game.addEffect(new PlotAddSpellAbilityEffect(new MageObjectReference(mainCard, game)), source);
-            game.informPlayers(owner.getLogName() + " plots " + mainCard.getLogName());
+            game.informPlayers(owner.getLogName() + " plots " + mainCard.getLogName() + " from " + zone.toString().toLowerCase());
```

Comment:
Must add source info at the end by  "xxx + CardUtil.getSourceLogName(game, source)"